src/commcare_cloud/ansible/roles/commcarehq/tasks/celery_cron.yml

```
@@ -1,27 +1,59 @@
-# To remove celery tasks that are running for more than 36 hours.
-#
+# To remove old celery tasks that are running from long time
+# SoftKill: SIGTERM, HardKill: SIGKILL
+
 - name: Celery cron jobs for soft kill tasks running from last 36h
   become: true
   cron:
     name: "Sending SIGTERM to celery processes running morethan 36h"
-    job: "kill -15 $(ps -eo comm,pid,etimes | awk '/celeryd:/ {if ($4 > 36*60*60) { print $3 }}')"
+    cron_file: celery_sigterm
+    state: absent
+  tags:
+    - cron
+
+- name: Cron job to Kill Celery tasks running from last 46h
+  become: true
+  cron:
+    name: "Sending SIGTERM to celery processes running morethan 46h"
+    cron_file: celery_sigkill
+    state: absent
+  tags:
+    - cron
+
+- name: Copy celery cron script
+  become: yes
+  template:
+    src: "celery_task_kill.sh.j2"
+    dest: "/usr/local/sbin/celery_task_kill.sh"
+    group: root
+    owner: root
+    mode: 0744
+    backup: yes
+  when: "celery_hours_before_soft_kill and celery_hours_before_hard_kill"
```

The tasks below only check one of the vars so the script needs to be present if either is set
```suggestion
  when: "celery_hours_before_soft_kill or celery_hours_before_hard_kill"
```